shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -741,6 +744,54 @@ public static TextDirection fromInt(int value) {
         }
     }
 
+    private class HintTextProxy {
+        private Method setHintText;
+        private Method setShowingHintText;
+
+        HintTextProxy() {
+            if (Build.VERSION.SDK_INT >= 26) {
+                Class<AccessibilityNodeInfo> nodeClass = AccessibilityNodeInfo.class;
+                try {
+                    setHintText = nodeClass.getDeclaredMethod("setHintText", CharSequence.class);
+                    setShowingHintText = nodeClass.getDeclaredMethod("setShowingHintText", boolean.class);
+                } catch (NoSuchMethodException exception) {
+                    setHintText = null;
+                    setShowingHintText = null;
+                } catch (SecurityException exception) {
+                    setHintText = null;
+                    setShowingHintText = null;
+                }
+            } else {
+                setHintText = null;
+                setShowingHintText = null;
+            }
+
+        }
+
+        private void setText(AccessibilityNodeInfo node, SemanticsObject object) {
+            if (setHintText == null) {
+                node.setText(object.getValueLabelHint());
+            } else {
+                String textValue = null;
+                try {
+                    String hintText = object.getHint();
```

Comment:
I think you'd want to use the "label" for `setHint` and the "value" for `setText`.

Although Android is not clean on the terminology, we should be. `SemanticsNode.hint` is defined as "A brief description of the result of performing an action on this node.", which is not really what setHint is used for.

Reply:
Fixed - though the semantics information coming from our text fields is not really correct then.  I'll need to track down what the expected order of the various labels+hints is and implement some custom merging logic to get the right behavior.